tests/baselines/reference/overloadedConstructorFixesInferencesAppropriately.errors.txt

```diff
@@ -0,0 +1,34 @@
+overloadedConstructorFixesInferencesAppropriately.ts(26,32): error TS2339: Property 'success' does not exist on type '{}'.
+
+
+==== overloadedConstructorFixesInferencesAppropriately.ts (1 errors) ====
+    interface Box<T> {
+        v: T;
+    }
+    
+    interface ErrorResult {
+        readonly error: true
+    }
+    
+    interface AsyncLoaderProps<TResult extends {}> {
+        readonly asyncLoad: () => Box<TResult>;
+        readonly children: (result: Exclude<TResult, ErrorResult>) => string;
+    }
+    
+    class AsyncLoader<TResult extends {}> {
+        constructor(props: string, context: any);
+        constructor(props: AsyncLoaderProps<TResult>);
+        constructor(...args: any[]) {}
+    }
+    
+    function load(): Box<{ success: true } | ErrorResult> {
+        return null as any;
+    }
+    
+    new AsyncLoader({
+        asyncLoad: load,
+        children: result => result.success as any,

```

Comment:
`result` would previously get a type fixed on it while checking the first overload (via `asyncLoader`), which'd the inform the expression checking during the second overload. Ideally inference should still work here (since a source for the better type still exists), but this might need some more changes to make work (like fine-tuning the expression checking order). That is probably the only -0 change I've seen caused by this thus far.